M365/Remove-MailboxExtendedProperty.ps1

```diff
@@ -0,0 +1,62 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+#Requires -Modules @{ ModuleName="ExchangeOnlineManagement"; RequiredVersion="3.4.0" }
+#Requires -Modules @{ ModuleName="Microsoft.Graph.Users"; RequiredVersion="2.24.0" }
+#Requires -Modules @{ ModuleName="Microsoft.Graph.Mail"; RequiredVersion="2.24.0" }
+
+<#
+.SYNOPSIS
+    Removes the extended property (aka named property) the message.
+
+.DESCRIPTION
+    Takes as input the result from Search-MailboxExtendedProperty.ps1 and removes the extended property from the message.
+
+.PARAMETER MessagesWithExtendedProperty
+    A list of mailbox items and their extended property, that are to be removed.
+
+.EXAMPLE
+    $mailboxExtendedProperty = Get-MailboxExtendedProperty -Identity fred@contoso.com | Where-Object { $_.PropertyName -like '*Some Pattern*' }
+    $messagesWithExtendedProperty = .\Search-MailboxExtendedProperty.ps1 -MailboxExtendedProperty $mailboxExtendedProperty
+    .\Remove-MailboxExtendedProperty.ps1 -MessagesWithExtendedProperty $messagesWithExtendedProperty
+
+#>
+param(
+    [Parameter(Mandatory = $true, Position = 0)]
```

Comment:
Because we are doing a remove here, we need to include a `ShouldProcess` within the script.

Reply:
Implemented SupportsShouldProcess